src/App.tsx

```diff
@@ -15,36 +15,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
```

Comment:
you don't need this state, just create some function that will handle sorting and reversing logic and call it 

```
const visibleGoods = getPreparedGoods(goodsFromServer, {
    sortField,
    reversed,
  });
``